tests/test_converter.py

```diff
@@ -454,6 +458,140 @@ def test_strikeout_word():
     assert newline_lf(markdown_output) == expected_markdown
 
 
+def test_annotated_spaces():
+    input = {
+        'object': 'block',
+        'id': 'b0e9b35d-1c97-49d6-944d-feb9beb6c83c',
+        'created_time': '2022-03-30T02:50:00.000Z',
+        'last_edited_time': '2022-03-30T03:25:00.000Z',
+        'created_by': {
+            'object': 'user',
+            'id': 'ad6d20a2-36e9-4b6d-a49a-7f71f8ba9af6'},
+        'last_edited_by': {
+            'object': 'user',
+            'id': 'ad6d20a2-36e9-4b6d-a49a-7f71f8ba9af6'},
+        'has_children': False,
+        'archived': False,
+        'type': 'paragraph',
+        'paragraph': {
+            'color': 'default',
+            'text': [
```

Comment:
This is a big object to read through to understand what the test is doing. What do you think about this idea:

Write a utility function that can generate the objects for us using an easier to read format. I've used this technique often in other tests:

```python
input = annotated_text([("this ", "b"), ("is ", "bi"), ("a", "I"), (" test", "s")])
```

I think you get the idea.

Also, although I didn't read through the test data carefully, I suspect you're testing a few different things all in one big test. I suspect you did this in part because the current test data fixture is so big it felt bad to copy it a few times for multiple tests. Once you've created this little function I showed you, you can split apart this big test into a few smaller tests that all have good function names. Each test should test one particular situation. You may need several of them to cover all the corner cases.

Let me know if any of this is unclear and I'd be happy to help.

Reply:
I'd also update the other tests to use this new tool, where appropriate.

Reply:
makes sense, I'll comment when it's done

Reply:
Very close, nice work! everything you've added looks great but please reread my comment carefully and be sure you addressed everything I brought up. Also, I think you can use tuple unpacking to in the for loop so that you can give block[0] and block[1] better names.